test/e2e/cloud.go

```diff
@@ -270,47 +280,77 @@ func (c *cloud) getKopsSecurityGroup(clusterName string) (string, error) {
 // In any case the security group is tagged kubernetes.io/cluster/$clustername
 // so filter using that and try to find a security group with "node" in it. If
 // no such group exists, use the first one in the response
-func (c *cloud) getEksSecurityGroup(clusterName string) (string, error) {
-	request := &ec2.DescribeSecurityGroupsInput{
-		Filters: []*ec2.Filter{
+func (c *cloud) getEksSecurityGroupId(clusterName string) (string, error) {
+	securityGroups, err := c.getFilteredSecurityGroups(
+		[]*ec2.Filter{
 			{
 				Name: aws.String("tag-key"),
 				Values: []*string{
 					aws.String(fmt.Sprintf("kubernetes.io/cluster/%s", clusterName)),
 				},
 			},
 		},
-	}
-
-	response, err := c.ec2client.DescribeSecurityGroups(request)
+	)
 	if err != nil {
 		return "", err
 	}
 
-	if len(response.SecurityGroups) == 0 {
-		return "", fmt.Errorf("no security group found for cluster %s", clusterName)
-	}
-
-	groupId := aws.StringValue(response.SecurityGroups[0].GroupId)
-	for _, sg := range response.SecurityGroups {
-		if strings.Contains(strings.ToLower(*sg.GroupName), "node") {
-			groupId = aws.StringValue(sg.GroupId)
+	securityGroupId := aws.StringValue(securityGroups[0].GroupId)
+	for _, securityGroup := range securityGroups {
+		if strings.Contains(strings.ToLower(*securityGroup.GroupName), "node") {
+			securityGroupId = aws.StringValue(securityGroup.GroupId)
 		}
 	}
 
-	return groupId, nil
+	return securityGroupId, nil
 }
 
-func (c *cloud) getSubnetIds(clusterName string) ([]string, error) {
-	request := &ec2.DescribeSubnetsInput{
-		Filters: []*ec2.Filter{
+func (c *cloud) getKopsSubnetIds(clusterName string) ([]string, error) {
+	return c.getFilteredSubnetIds(
+		[]*ec2.Filter{
 			{
 				Name: aws.String("tag-key"),
 				Values: []*string{
 					aws.String(fmt.Sprintf("kubernetes.io/cluster/%s", clusterName)),
 				},
 			},
 		},
+	)
+}
+
+func (c *cloud) getEksSubnetIds(clusterName string) ([]string, error) {
+	return c.getFilteredSubnetIds(
+		[]*ec2.Filter{
+			{
+				Name: aws.String("tag:Name"),
+				Values: []*string{
+					aws.String(fmt.Sprintf("*%s*", clusterName)),
```

Comment:
Are all the eks clusters created with eksctl?

Reply:
Another problem is that you may get more subnets than expected.  In the event that you have `cluster1` and `cluster10`, you will get additional subnets when looking for those that belong to `cluster1`.

Reply:
Updated:

first, call getEksctlSubnetIds. eksctl gives us reliable way to find the subnets "tag:alpha.eksctl.io/cluster-name"

second, call getEksCloudFormationSubnetIds. this is necessarily very generic since the name could be anything but for now i narrowed it to "%s-*", clusterName because that's how **I** name mine